src/App.tsx

```diff
@@ -1,14 +1,75 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
 
+import { Todo } from './types/Todo';
+
+type Params = {
+  query: string;
+  filter: string;
+};
+
+function filterTodos(list: Todo[], params: Params) {
+  const { query, filter } = params;
+  let result: Todo[] = [];
+
+  if (filter) {
+    switch (filter) {
+      case 'all':
+        result = list;
+        break;
+
+      case 'active':
+        result = list.filter((todo: Todo) => !todo.completed);
+        break;
+
+      case 'completed':
+        result = list.filter((todo: Todo) => todo.completed);
+        break;
+
+      default:
+        result = list;
+        break;
+    }
+  }
+
+  if (query) {
+    result = result.filter((todo: Todo) =>
+      todo.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+
+  return result;
+}
```

Comment:
The `filterTodos` function uses a `filter` parameter, but the switch statement does not handle the case where `filter` is 'All'. This could lead to unexpected behavior since the initial state for `filter` is set to 'All'. Consider adding a case for 'All' or adjusting the initial state to match one of the handled cases.